python/tests/conftest.py

```diff
@@ -271,9 +408,18 @@ def mongo_version_store(mongo_store_factory):
     return mongo_store_factory()
 
 
-@pytest.fixture(scope="function")
-def s3_version_store_prune_previous(s3_store_factory):
-    return s3_store_factory(prune_previous_version=True)
+@pytest.fixture(
+    scope="function",
+    params=["s3_store_factory", "azure_store_factory"] if AZURE_SUPPORT else ["s3_store_factory"],
```

Comment:
Instead of doing it here, create a new fixture `object_store_factory` with this parameterization. You can then reuse that in most places you're checking `AZURE_SUPPORT`.